### PR TITLE
Add sample projects

### PR DESCRIPTION
**Pull request by Shane32**
Prep for new documentation.

Includes sample projects as follows:

| Name | Framework | Description |
|-|-|-|
| Authorization | .NET 6 Minimal | Based on the VS template, demonstrates authorization functionality with cookie-based authentication |
| Basic | .NET 6 Minimal | Simplest possible implementation |
| Complex | .NET 3.1 / 5 / 6 | Rename of existing sample server, which demonstrates older Program/Startup files and various configuration options, and multiple UI endpoints |
| Controller | .NET 6 Minimal | MVC implementation; does not include WebSocket support |
| Cors | .NET 6 Minimal | Demonstrates CORS setup |
| EndpointRouting | .NET 6 Minimal | Demonstrates configuring GraphQL through endpoint routing |
| MultipleSchemas | .NET 6 Minimal | Demonstrates configuring multiple schemas within a single server |
| Net48 | .NET Core 2.1 / .NET 4.8 | Demonstrates configuring GraphQL on .NET 4.8 / Core 2.1 |
| Pages | .NET 6 Minimal | Demonstrates configuring GraphQL on top of a Razor Pages template |

All sample projects include very basic tests, testing the UI endpoint, and basic GET / POST / WebSocket connectivity.

Since the projects are based on .NET templates, some contain a bit of excess code.  Notably the Authorization sample project has quite a bit of auto-generated code.  This is "by design", in that it demonstrates to users how to integrate GraphQL.NET with the standard .NET templates.  On the other hand, a few of the samples are just a couple files.

There is also a few copies of the GraphiQL HTML page, for a similar reason.

We could consider putting these in the Examples repo also, but I personally like them here....

### CODE REVIEW

File: samples/Samples.MultipleSchemas/Pages/Index.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+using Microsoft.AspNetCore.Mvc;
```
Comment (Shane32) [inline samples/Samples.MultipleSchemas/Pages/Index.cshtml.cs:1]:
```suggestion
```

**Comment by sungam3r** [comment]
> We could consider putting these in the Examples repo also, but I personally like them here....

Or absorb examples repo here and archive it.

**Comment by sungam3r** [comment]
There is no CI in examles, no checks at all.

**Comment by Shane32** [comment]
On a side note, I would really like to add one more sample project here demonstrating OAuth & JWT bearer authentication with a SPA website, with the focus on OAuth/JWT authentication and full support of authentication for queries, mutations and subscriptions over an authenticated channel.  I anticipate that IdentityServer4 (or the newer replacement) can adequately handle OAuth and the sample can demonstrate how GraphQL.NET Server validates JWT-authenticated requests.  Since authentication over WebSockets is tricky when being called from a browser, we should be sure to cover that as well.  I feel that the above sample will be extremely valuable from a documentation point of view.  (This code is not ready and would be a separate PR after the release of v7.)

File: samples/Samples.Authorization/Program.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// ------ added code for GraphQL -------
+app.UseWebSockets();
+// configure the graphql endpoint at "/graphql"
+app.UseGraphQL("/graphql");
+// configure Playground at "/ui/graphql"
+app.UseGraphQLPlayground(
+    new GraphQL.Server.Ui.Playground.PlaygroundOptions
+    {
+        GraphQLEndPoint = new PathString("/graphql"),
+        SubscriptionsEndPoint = new PathString("/graphql"),
```
Comment (sungam3r) [inline samples/Samples.Authorization/Program.cs:68]:
```suggestion
        GraphQLEndPoint = "/graphql",
        SubscriptionsEndPoint = "/graphql",
```

File: samples/Samples.Basic/Program.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+app.UseDeveloperExceptionPage();
+app.UseWebSockets();
+// configure the graphql endpoint at "/graphql"
+app.UseGraphQL("/graphql");
+// configure Playground at "/"
+app.UseGraphQLPlayground(
+    new GraphQL.Server.Ui.Playground.PlaygroundOptions
+    {
+        GraphQLEndPoint = new PathString("/graphql"),
+        SubscriptionsEndPoint = new PathString("/graphql"),
```
Comment (sungam3r) [inline samples/Samples.Basic/Program.cs:23]:
```suggestion
        GraphQLEndPoint = "/graphql",
        SubscriptionsEndPoint = "/graphql",
```

File: samples/Samples.Controller/Controllers/HomeController.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        }
+        else if (HttpContext.Request.HasJsonContentType())
+        {
+            var request = await _serializer.ReadAsync<GraphQLRequest>(HttpContext.Request.Body, HttpContext.RequestAborted);
+            return await ExecuteGraphQLRequestAsync(request);
+        }
+        return BadRequest();
+    }
+
+    private GraphQLRequest ParseRequest(string query, string? operationName, string? variables = null, string? extensions = null)
```
Comment (sungam3r) [inline samples/Samples.Controller/Controllers/HomeController.cs:57]:
```suggestion
    private GraphQLRequest BuildRequest(string query, string? operationName, string? variables = null, string? extensions = null)
```

File: samples/Samples.MultipleSchemas/Program.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+// configure the graphql endpoint at "/cats/graphql"
+app.UseGraphQL<MultipleSchema.Cats.CatsSchema>("/cats/graphql");
+// configure the graphql endpoint at "/dogs/graphql"
+app.UseGraphQL<MultipleSchema.Dogs.DogsSchema>("/dogs/graphql");
+// configure Playground at "/cats"
+app.UseGraphQLPlayground(
+    new GraphQL.Server.Ui.Playground.PlaygroundOptions
+    {
+        GraphQLEndPoint = new PathString("/cats/graphql"),
+        SubscriptionsEndPoint = new PathString("/cats/graphql"),
```
Comment (sungam3r) [inline samples/Samples.MultipleSchemas/Program.cs:24]:
```suggestion
        GraphQLEndPoint = "/cats/graphql",
        SubscriptionsEndPoint = "/cats/graphql",
```

File: samples/Samples.MultipleSchemas/Program.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    },
+    "/cats");
+// configure Playground at "/dogs"
+app.UseGraphQLPlayground(
+    new GraphQL.Server.Ui.Playground.PlaygroundOptions
+    {
+        GraphQLEndPoint = new PathString("/dogs/graphql"),
+        SubscriptionsEndPoint = new PathString("/dogs/graphql"),
```
Comment (sungam3r) [inline samples/Samples.MultipleSchemas/Program.cs:32]:
```suggestion
        GraphQLEndPoint = "/dogs/graphql),
        SubscriptionsEndPoint = "/dogs/graphql",
```

**Comment by sungam3r** [comment]
It would be nice to add a table from PR summary into Readme.

**Comment by Shane32** [comment]
> It would be nice to add a table from PR summary into Readme.

That will be part of the documentation update, which this PR is prep for.

**Review by sungam3r — APPROVED** [review]

**Pull request merged**